driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/internal/BatchCursorFluxTest.java

```diff
@@ -299,6 +302,30 @@ void changeStreamPublisherCompletesAfterDroppingCollection() {
         subscriber.assertNoErrors();
     }
 
+    @Test
+    @DisplayName("Ensure BatchCursor does not drop an error")
+    public void testBatchCursorDoesNotDropAnError() {
+        try {
+            AtomicBoolean errorDropped = new AtomicBoolean();
+            Hooks.onErrorDropped(t -> errorDropped.set(true));
+            Mono.from(collection.insertMany(createDocs(200))).block();
+
+            Flux.fromStream(IntStream.range(1, 200).boxed())
```

Comment:
I guess this test is timing-dependent?

Reply:
Yes and resource dependent.  

Essentially, this test forces 400 queries to the collection but abandons 50% of the cursors as one of the queries returns a result quicker than the other. This ends up triggering the race consistently.